src/core/ref.h

```diff
@@ -49,9 +50,21 @@ static inline uint32_t ref_dec(Ref* ref) { return atomic_fetch_sub(ref, 1) - 1;
 
 #endif
 
+typedef bool (*RefDestructor)(void* o, void *extra);
+typedef struct RefHeader {
+  Ref ref;
+  RefDestructor destructor;
+  void* destructorExtra;
+} RefHeader;
```

Comment:
This is a pretty big change to the object system, so I just want to make sure that there is a good reason to go from 4 -> 24 bytes.

More background: Refs used to store their destructor/name until I realized that it was a lot of redundant information and the only place we truly needed "polymorphic" information was for Variants.  In 99% of places we already know what destructor to use when releasing the object.

Reply:
Yeah. I saw _lovrRelease but special casing release everywhere that a SoundData might be used felt very error prone to me. Can you think of another approach? I couldn't :S 

Also, it's 8 bytes; size_t was used for malloc even though the payload was only 4 bytes.